PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1

```diff
@@ -710,6 +710,40 @@ function Start-CopyJob(
     }
 }
 
+#
+# Utility wrapper for invoking commands by opening sessions 
+# for each of the cluster nodes and preserving the session 
+# to be deleted after use.
+#
+
+function New-InvokeCommand (
+    [string[]] $ClusterNodes = @(),
+    [string] $JobName,
+    [scriptblock] $ScriptBlock
+    )
+{
+	$Jobs       = @()
+	$Sessions   = @()
+	$SessionIds = @()
+
+	if ($ClusterNodes.Count -ge 1) {

```

Comment:
what if -eq 1? I agree validation could be good, suggest doing that explicitly up front. I believe making ClusterNodes a mandatory parameter would enforce non-null (which is what you want).

https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_functions_advanced_parameters?view=powershell-6

Reply:
Sorry, meant -eq 0 ...

Reply:
Agreed.